my-app/src/Login.js

```diff
@@ -95,10 +95,11 @@ export default class Login extends Component {
   }
 
   //----------Checks if the user is previously logged in
-  componentDidMount() {
+  componentWillMount() {
```

Comment:
It seems like you have a better idea of the difference between componentDidMount and componentWillMount. Could you add a brief comment to explain the difference?

Reply:
So from what I know, componentWillMount will be called by both server and client before the render() is executed, while componentDidMount is called after the render is executed at least once. From what I heard, the codes for initializing state is better to be put inside of componendDidMount; so basically I didn't put those code intentionally (nor I remember changing it, perhaps an accident from pasting). Apparently, we also already have componentDidMount() at line 24. I can just dump this code into there, what do you think?

Reply:
So from what I know, componentWillMount will be called by both server and client before the render() is executed, while componentDidMount is called after the render is executed at least once. From what I heard, the codes for initializing state is better to be put inside of componendDidMount; so basically I didn't put those code intentionally (nor I remember changing it, perhaps an accident from pasting). Apparently, we also already have componentDidMount() at line 24. I can just dump this code into there, what do you think?

Reply:
Just to re-clarify for the reason why I put the code inside of componentWillMount.
componentWillMount: will execute the codes BEFORE the page renders.
componentDidMount: will execute the codes AFTER the page renders.
Since we want to know whether the user has logged in or not BEFORE the page renders what it suppose to render, it should be in componentWillMount.